### ISSUE TITLE
Unicode issue when sending with python 3.11.9, 3.12.3

### ISSUE THREAD

**Issue by blighj** [django-ses/django-ses#315]
It looks like a change added in python 3.11.9 [#76511 Email parser creates a message object that can't be flattened](https://github.com/python/cpython/issues/76511) has cuased a bug in django-ses.
If you send a message with unicode in the body you get the following error

```
File "/usr/local/lib/python3.11/site-packages/django/core/mail/message.py", line 169, in <genexpr>
    len(line.encode()) > RFC5322_EMAIL_LINE_LENGTH_LIMIT
        ^^^^^^^^^^^^^
UnicodeEncodeError: 'utf-8' codec can't encode characters in position 3-5: surrogates not allowed
```

You can replicate this by updating the test_send_mail https://github.com/django-ses/django-ses/blob/83b32eca995b7eb495395d860a61a44533e19e5a/tests/test_backend.py#L115-L127

to
```diff
    def test_send_mail(self):
        settings.AWS_SES_CONFIGURATION_SET = None

        unicode_from_addr = 'Unicode Name óóóóóó <from@example.com>'

-       send_mail('subject', 'body', unicode_from_addr, ['to@example.com'])
+       send_mail('subject', 'body with unicode óóóóóó', unicode_from_addr, ['to@example.com'])
        message = self.outbox.pop()
        mail = email.message_from_string(smart_str(message['RawMessage']['Data']))
        self.assertTrue('X-SES-CONFIGURAITON-SET' not in mail.keys())
        self.assertEqual(mail['subject'], 'subject')
        self.assertEqual(mail['from'], self._rfc2047_helper(unicode_from_addr))
        self.assertEqual(mail['to'], 'to@example.com')
-       self.assertEqual(mail.get_payload(), 'body')
+       self.assertEqual(mail.get_payload(), 'body with unicode óóóóóó')
```

This test will pass under python 3.11.8 but fail under 3.11.9

**Comment by pcraciunoiu** [comment]
@blighj @tudoramariei thanks for filing/referencing. If you would be willing to contribute to fix this issue, I'd be glad to take a look at the PR

**Comment by michael-agel-ambient** [comment]
Hi there, I just flew by to let you know that Python 3.12.3 also seems to be affected by this. On Python 3.12.2, everything looks good to me. This probably also goes for the latest Python 3.13, but I haven't examined that.

**Comment by GitRon** [comment]
@pcraciunoiu I fixed the bug! Would be awesome if you could have a look (and release) quite soon. Thx ❤️

**Comment by pcraciunoiu** [comment]
This is released on PyPI as v3.6.0 - https://pypi.org/project/django-ses/3.6.0/

**Comment by GitRon** [comment]
@pcraciunoiu You are the best, thx for the super-fast release ❤️
